### PR TITLE
hwmv2: Port STM32MP1X SoCs & boards

### PR DESCRIPTION
**Pull request by ajarmouni-st**
Port STM32MP1X SoCs & boards to Hardware Model V2.

### CODE REVIEW

**Review by erwango — COMMENTED** [review]
Despite inforamtion coming from yaml, I wonder if vendor should actually be 96boards instead of arrow (see https://www.96boards.org/product/avenger96/)

**Comment by ajarmouni-st** [comment]
> Despite inforamtion coming from yaml, I wonder if vendor should actually be 96boards instead of arrow (see https://www.96boards.org/product/avenger96/)

I agree.

File: soc/st/stm32/soc.yml [hunk 1]
```diff
@@ -21,3 +21,6 @@ family:
     - name: stm32f103xe
     - name: stm32f105xc
     - name: stm32f107xc
+  - name: stm32mp1x
+    socs:
+    - name: stm32mp157cxx
```
Comment (erwango) [inline soc/st/stm32/soc.yml:26]: If A core is ported once day, this will have to be fixed, but it should be good for now

**Review by erwango — APPROVED** [review]

**Review by arnopo — APPROVED** [review]
LGTM

**Comment by ajarmouni-st** [comment]
Rebased to resolve conflict.

**Comment by ajarmouni-st** [comment]
> ajarmouni-st requested review from arnopo and erwango now

By mistake, sorry!

**Review by erwango — APPROVED** [review]
Soc part will have to be updated if Cortex A is supported one day, but in current state it is ok.

**Review by arnopo — APPROVED** [review]

**Comment by nordicjm** [comment]
rebase needed

**Review by nordicjm — APPROVED** [review]

**Comment by nordicjm** [comment]
rebase needed

**Pull request merged**